### PR TITLE
HM7

### PR DESCRIPTION
**Pull request by IuliiaMazhara**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw7/enteties/MetalColor.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    public List<String> getVegetables() {
+        return vegetables;
+    }
+
+    public void setSummary(List<String> summary) {
+        this.summary = summary;
+    }
+
+    public List<String> getActualResultLog() {
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/enteties/MetalColor.java:62]: I think this methods provides not actual, but expected List of Strings
If it does, then could you please rename method

File: src/test/java/com/epam/tc/hw7/enteties/MetalColor.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        return actualResult;
+    }
+
+    private static String convertStringArrayToString(List<String> strArr, String delimiter) {
+        StringBuilder sb = new StringBuilder();
+        for (String str : strArr) {
+            sb.append(str).append(delimiter).append(' ');
+        }
+        return sb.substring(0, sb.length() - 2);
+    }
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/enteties/MetalColor.java:81]: If you want, this method can be replaced by `java.lang.String#join(java.lang.CharSequence, java.lang.Iterable<? extends java.lang.CharSequence>)` that does almost the same under the hood.

Anyway, your implementation is good :+1:

File: src/test/java/com/epam/tc/hw7/enteties/User.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.epam.tc.hw7.enteties;
+
+
+public class User {
+    public static final  User ROMAN = new User("Roman", "Jdi1234", "ROMAN IOVLEV");
+
+    private String userName;
+    private String password;
+    private String fullName;
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/enteties/User.java:9]: As improvement: Please add `final` modified to these fields

File: src/test/java/com/epam/tc/hw7/enteties/User.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package com.epam.tc.hw7.enteties;
+
+
+public class User {
+    public static final  User ROMAN = new User("Roman", "Jdi1234", "ROMAN IOVLEV");
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/enteties/User.java:5]: In general, immutable data transfer object (immutable DTO, Value object) should not be used to store such constants.
It is not so important in this homework (you can resolve this comment), but as improvement please move it to different place, e.g. dedicated class

File: src/test/java/com/epam/tc/hw7/enteties/MetalColor.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
+package com.epam.tc.hw7.enteties;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MetalColor {
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/enteties/MetalColor.java:7]: As improvement you can extend `com.jdiai.tools.DataClass` like 
```suggestion
public class MetalColor extends DataClass<MetalColor> {
```
benefits of this approach can be found in [documentaton](https://jdi-docs.github.io/jdi-light/#standard-login-form)

\+ same for `User` class

File: src/test/java/com/epam/tc/hw7/enteties/MetalColor.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
+package com.epam.tc.hw7.enteties;
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/enteties/MetalColor.java:1]: minor typo 
```suggestion
package com.epam.tc.hw7.entities;
```

File: src/test/java/com/epam/tc/hw7/MetalColorPage.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    public void fillMetalAndColorForm(MetalColor metalColor) {
+        metalColorForm.fill(metalColor);
+    }
+
+    public void assertResultLog(MetalColor metalColor) {
+        Assertions.assertThat(metalColor.getActualResultLog())
+                .hasSameElementsAs(expectedActualResultLog());
+    }
+
+    public List<String> expectedActualResultLog() {
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/MetalColorPage.java:29]: I think this method return actual state (not expected).
Could you please rename it accordingly

File: src/test/java/com/epam/tc/hw7/MetalColorPage.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import com.epam.jdi.light.elements.composite.WebPage;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.Css;
+import com.epam.tc.hw7.enteties.MetalColor;
+import com.epam.tc.hw7.forms.MetalColorForm;
+import java.util.ArrayList;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+
+
+public class MetalColorPage extends WebPage {
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/MetalColorPage.java:13]: I think `pages` package could be a better place for this class
\+ JdiSite.java too

File: src/test/java/com/epam/tc/hw7/DataProviderMetalColor.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import org.testng.annotations.DataProvider;
+
+
+public class DataProviderMetalColor {
+
+    private static final String path = "src/test/resources/JdiDataSet.json";
+
+    Object[][] object;
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/DataProviderMetalColor.java:18]: This is candidate to be local variable

File: src/test/java/com/epam/tc/hw7/DataProviderMetalColor.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    public Object[][] dataProviderMetalColor() {
+
+        {
+            ObjectMapper mapper = new ObjectMapper();
+            FileInputStream fileInputStream = null;
+            try {
+                fileInputStream = new FileInputStream(path);
+            } catch (FileNotFoundException e) {
+                throw new RuntimeException(e);
+            }
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw7/DataProviderMetalColor.java:30]: Please use try-with-resources and move code below inside this try

**Review by shoggy — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7/enteties/MetalColor.java (hunk 1 above)
Comment (IuliiaMazhara) [reply src/test/java/com/epam/tc/hw7/enteties/MetalColor.java:62]: Renamed the method

File: src/test/java/com/epam/tc/hw7/enteties/User.java (hunk 3 above)
Comment (IuliiaMazhara) [reply src/test/java/com/epam/tc/hw7/enteties/User.java:9]: Added final

File: src/test/java/com/epam/tc/hw7/enteties/MetalColor.java (hunk 6 above)
Comment (IuliiaMazhara) [reply src/test/java/com/epam/tc/hw7/enteties/MetalColor.java:1]: Corrected type

File: src/test/java/com/epam/tc/hw7/DataProviderMetalColor.java (hunk 9 above)
Comment (IuliiaMazhara) [reply src/test/java/com/epam/tc/hw7/DataProviderMetalColor.java:18]: Moved from class field to local variable

File: src/test/java/com/epam/tc/hw7/DataProviderMetalColor.java (hunk 10 above)
Comment (IuliiaMazhara) [reply src/test/java/com/epam/tc/hw7/DataProviderMetalColor.java:30]: Refactored data ptovider

File: src/test/java/com/epam/tc/hw7/MetalColorPage.java (hunk 8 above)
Comment (IuliiaMazhara) [reply src/test/java/com/epam/tc/hw7/MetalColorPage.java:13]: Moved to pages folder

File: src/test/java/com/epam/tc/hw7/enteties/MetalColor.java (hunk 5 above)
Comment (IuliiaMazhara) [reply src/test/java/com/epam/tc/hw7/enteties/MetalColor.java:7]: Extended, read about benefits

File: src/test/java/com/epam/tc/hw7/enteties/MetalColor.java (hunk 2 above)
Comment (IuliiaMazhara) [reply src/test/java/com/epam/tc/hw7/enteties/MetalColor.java:81]: used String.join

**Review by shoggy — APPROVED** [review]

**Pull request merged**